private/association-list.rkt

```diff
@@ -97,7 +97,7 @@
   (hash-ref (association-list-backing-hash assoc) k empty-immutable-vector))
 
 (define (association-list-keys assoc)
-  (list->multiset
+  (sequence->multiset
```

Comment:
This can skip the `sequence->multiset` step completely by using `for*/multiset` instead of `for*/list`. The only reason this code uses `for*/list` in the first place is because when I first wrote it, I hadn't implemented `for*/multiset` yet.

Reply:
Done. Thanks!